XenAdmin/Wizards/ImportWizard/NetworkPickerPage.cs

```diff
@@ -274,7 +274,9 @@ private void AddVIFRow(VIF vif)
 		{
             var row = new VifRow(vif);
             XenAPI.Network network = m_selectedConnection.Resolve(vif.network);
-            bool isGuestInstallerNetwork = network != null && network.IsGuestInstallerNetwork;
+            bool isGuestInstallerNetwork = network != null &&
+                network.IsGuestInstallerNetwork && !XenAdmin.Properties.Settings.Default.ShowHiddenVMs;
+            // CA-218956 - Expose HIMN when showing hidden objects

```

Comment:
The check for hidden objects shouldn't be done here, but on line 300 below 
```
if (isGuestInstallerNetwork)
    row.Visible = false;
```
The logic here is that we should always add a row for the guest installer network, to ensure that the corresponding VIF gets created; and we should then hide it, **if hidden objects are not shown**.